src/app/catalog/product/components/product-detail/product-detail.component.ts

```diff
@@ -27,4 +29,10 @@ export class ProductDetailComponent {
   showConfirmation() {
     this.store.dispatch(navigate({ url: '/order' }));
   }
+
+  ngOnInit() {
+    const productId = Number.parseInt(this.route.snapshot.paramMap.get('id'));
+    this.store.dispatch(loadProducts());
```

Comment:
Why do we need to load all products when opening a single product?

Reply:
removed